airbyte-integrations/connectors/destination-s3/src/main/java/io/airbyte/integrations/destination/s3/parquet/JsonFieldNameUpdater.java

```diff
@@ -0,0 +1,80 @@
+/*
+ * MIT License
+ *
+ * Copyright (c) 2020 Airbyte
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy
+ * of this software and associated documentation files (the "Software"), to deal
+ * in the Software without restriction, including without limitation the rights
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in all
+ * copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+ * SOFTWARE.
+ */
+
+package io.airbyte.integrations.destination.s3.parquet;
+
+import com.fasterxml.jackson.databind.JsonNode;
+import com.google.common.collect.ImmutableMap;
+import io.airbyte.commons.json.Jsons;
+import java.util.Map;
+
+/**
+ * This helper class tracks whether a Json has special field name that needs to be replaced with a
```

Comment:
Parquet only allows these characters for record name: `/a-zA-Z0-9_/`. Otherwise I won't go through all the trouble to do this. The necessity of this tracker actually complicates things a lot.

Will update the comment to reflect that.

Reply:
We currently have to deal with some naming conventions because some destinations allow different subsets of characters for identifiers names. These are dealt in classes deriving from `airbyte-integrations/bases/base-java/src/main/java/io/airbyte/integrations/destination/NamingConventionTransformer.java`

It seems that for S3 parquet, a difference would be that it needs to apply those conventions to field names too. Other destinations only cares about conventions for stream names and namespace (table & schema).

But would it make sense to regroup them in the same kind of class/hierarchy too?

Reply:
> But would it make sense to regroup them in the same kind of class/hierarchy too?

@ChristopheDuong, can you elaborate on this? What do you mean by "regroup them in the same kind of class"?

Reply:
> @ChristopheDuong, can you elaborate on this? What do you mean by "regroup them in the same kind of class"?

Should we move this code to a class named `S3NameTransformer` that extends `NamingConventionTransformer` like we do with `SnowflakeSQLNameTransformer` or `RedshiftSQLNameTransformer` etc ?

Reply:
I see.

The name conversion logic for Parquet is exactly the same as the one in `StandardNameTransformer`. So there is nothing to override. It seems unnecessary to create a new class.